glossary.rst

```diff
@@ -0,0 +1,151 @@
+      some more auxiliary files to build or run the project such as generators for different build
+      systems or files to be imported from the package to the local folder.
+
+   recipe
+      Python script defined in a `conanfile.py` that specifies how the package is built from sources,
+      what the final binary artifacts are, the package dependencies, etc.
+
+   reference
+      A package reference is the combination of the package name, version, and two optional fields
+      named user and channel that could be useful to identify a forked recipe from the community with
+      changes specific to your company. It is show in the the form of: `<name>/<version>@<user>/<channel>`.
+
+   package
+      A Conan package is a collection of files meant to be consumed for a certain configuration and
+      settings. It can contain binary files such as libraries, headers or tools to be reused by the
+      consumer of the package.
+
+   lockfile
+      Files that store the information with the exact versions, revisions, options, and configuration
+      of a dependency graph. They are intended to make the building process reproducible even if the
+      dependency definitions in conanfile recipes are not fully deterministic.
+
+   revision
+      Is the way to implicitly version the changes done in a recipe or package without bumping the
+      actual reference or package version.
+
+   recipe revision
+      A unique ID using the latest VCS hash or a checksum of the recipe manifest (`conanfile.py` with
+      files exported if any).
+
+   package revision
+      A unique ID using the checksum of the package manifest (all files stored in a binary package).
+
+   package ID
+      The package id is a hash of the settings options and requirements used to identify the binary
+      packages
+
+   binary package
+      Output binary for the recipe compatible for certain settings and options.
+
+   settings
+      Settings are a set of different configurations that define the ABI of your package and that a 
+      consumer can indicate as input to retrieve the correct packages for his configuration.
+
+   options
+      An option is something that can be defaulted by the package creator, like if a library is
+      static or shared. Options are specific to each package, and each package creator can define
+      their options "header_only" for example. There is an option that is the most usual, and
+      recommended name shared = True/False
+
+   requirement
+      Packages on which another package depends on.
+
+   build requirement
+      Requirements that are only needed when you need to build a package from sources, but if the
+      binary package already exists, you don’t want to install or retrieve them.
+
+   dependency
+      A component that is directly referenced by a program.
+
+   dependency graph
+      A directed graph representing dependencies of several Conan packages towards each other.
```

Comment:
```suggestion
      A directed graph representing dependencies of several Conan packages towards each other. The relations between the packages are declared with the `requirements` in the recipes. A dependency graph in Conan depends on the input profile applied because the requirements can be conditioned to a specific configuration.
```
LINK TO METHOD REQUIREMENTS OR WHERE WE DECLARE CONDITIONAL REQUIRES